screen2.0/src/app/search/page.tsx

```diff
@@ -1,91 +1,517 @@
-    distancePC: searchParams.distancePC ? checkTrueFalse(searchParams.distancePC) : true,
-    distanceAll: searchParams.distanceAll ? checkTrueFalse(searchParams.distanceAll) : true,
-    distanceFromcCRE: searchParams.distanceFromcCRE ? Number(searchParams.distanceFromcCRE) : 1000000,
-    CTCF_ChIA_PET: searchParams.CTCF_ChIA_PET ? checkTrueFalse(searchParams.CTCF_ChIA_PET) : true,
-    RNAPII_ChIA_PET: searchParams.RNAPII_ChIA_PET ? checkTrueFalse(searchParams.RNAPII_ChIA_PET) : true
+      else setPage(page - 1)
+    }
+    //No action needed when closing a tab to the right of the page you're on
+  }
+
+  const handlePageChange = (_, newValue: number) => {
+    if (mainQueryParams.gene.name && (newValue === 2 || newValue === 3)) {
+      setOpen(false)
+    }
+    setPage(newValue)
   }
 
-  //Contains cell type data of the specified assembly
-  const globals: CellTypeData = await getGlobals(mainQueryParams.assembly)
+  //Keep URL and state in sync. Prevent from firing initially to allow time for opencCREs to be initialized
+  useEffect(() => {
+    //Check if the URL params representing state are stale
+    if (
+      opencCREsInitialized &&
+      JSON.stringify(constructMainQueryParamsFromURL(searchParams)) !== JSON.stringify(mainQueryParams)
+      || JSON.stringify(constructFilterCriteriaFromURL(searchParams)) !== JSON.stringify(filterCriteria)
+      || JSON.stringify(constructBiosampleTableFiltersFromURL(searchParams)) !== JSON.stringify(biosampleTableFilters)
+      || +searchParams.page !== page
+      || searchParams.accessions !== opencCREs.map(x => x.ID).join(','))
+     {
+      const newURL = constructSearchURL(
+        mainQueryParams,
+        filterCriteria,
+        biosampleTableFilters,
+        page,
+        opencCREs.map(x => x.ID).join(',')
+      )
+      router.push(newURL)
+    }
+  }, [searchParams, mainQueryParams, filterCriteria, biosampleTableFilters, page, opencCREs, router, basePathname, opencCREsInitialized])
+
+  //fetch globals
+  useEffect(() => {
+    // Setting react/experimental in types is not fixing this error? https://github.com/vercel/next.js/issues/49420#issuecomment-1537794691
+    // @ts-expect-error
+    startTransition(async () => {
+      console.log("fetching globals")
+      setGlobals(await getGlobals(mainQueryParams.coordinates.assembly))
+    })
+  }, [mainQueryParams.coordinates.assembly])
+
+
+  //Fetch raw cCRE data (main query and linked genes)
+  useEffect(() => {
+      setLoadingFetch(true)
+
+      let start: number;
+      if (mainQueryParams.snp.rsID) {
+        start = Math.max(0, mainQueryParams.coordinates.start - mainQueryParams.snp.distance);
+      } else if (mainQueryParams.gene.nearTSS) {
+        start = TSSs ? Math.max(0, Math.min(...TSSs) - mainQueryParams.gene.distance) : null
+      } else {
+        start = mainQueryParams.coordinates.start;
+      }
+
+      let end: number;
```

Comment:
we can just assign end with mainQueryParams.coordinates.end initially and else block can be removed

Reply:
fixed